geospaas_rest_api/filters.py

```diff
@@ -1,111 +1,256 @@
+                 'filter_class': CharFilter
+             }
+        }
+
+
+class SourceFilter(StrictFilterSet):
+    """Filter for Sources"""
+    instrument = rest_framework_filters.RelatedFilter(
+        InstrumentFilter,
+        field_name='instrument',
+        queryset=geospaas.vocabularies.models.Instrument.objects.all()
+    )
+    platform = rest_framework_filters.RelatedFilter(
+        PlatformFilter,
+        field_name='platform',
+        queryset=geospaas.vocabularies.models.Platform.objects.all()
+    )
+
+
+class PersonnelFilter(StrictFilterSet):
+    """Filter for Personnel objects"""
+    class Meta:
+        model = geospaas.catalog.models.Personnel
+        fields = {
+            'phone': '__all__',
+            'fax': '__all__',
+            'address': '__all__',
+            'city': '__all__',
+            'province_or_state': '__all__',
+            'postal_code': '__all__',
+            'country': '__all__'
+        }
+
+
+class RoleFilter(StrictFilterSet):
+    """Filter for Roles"""
+    personnel = rest_framework_filters.RelatedFilter(
+        PersonnelFilter,
+        field_name='personnel',
+        queryset = geospaas.catalog.models.Personnel.objects.all()
+    )
+    class Meta:
+        model = geospaas.catalog.models.Role
+        fields = {'role': '__all__'}
+
+
+class DatasetFilter(StrictFilterSet):
+    """Filter for Datasets"""
+    iso_topic_category = rest_framework_filters.RelatedFilter(
+        ISOTopicCategoryFilter,
+        field_name='ISO_topic_category',
+        queryset=geospaas.vocabularies.models.ISOTopicCategory.objects.all()
+    )
+    data_center = rest_framework_filters.RelatedFilter(
+        DataCenterFilter,
+        field_name='data_center',
+        queryset=geospaas.vocabularies.models.DataCenter.objects.all()
+    )
+    source = rest_framework_filters.RelatedFilter(
+        SourceFilter,
+        field_name='source',
+        queryset=geospaas.catalog.models.Source.objects.all()
+    )
+    geographic_location = rest_framework_filters.RelatedFilter(
+        GeographicLocationFilter,
+        field_name='geographic_location',
+        queryset=geospaas.catalog.models.GeographicLocation.objects.all()
+    )
+    gcmd_location = rest_framework_filters.RelatedFilter(
+        LocationFilter,
+        field_name='gcmd_location',
+        queryset=geospaas.vocabularies.models.Location.objects.all()
+    )
+    parameters = rest_framework_filters.RelatedFilter(
+        ParameterFilter,
+        field_name='parameters',
+        queryset=geospaas.vocabularies.models.Parameter.objects.all()
+    )
+
+    class Meta:
+        model = geospaas.catalog.models.Dataset
+        fields = {
+            'id': '__all__',
+            'entry_id': '__all__',
+            'entry_title': '__all__',
+            'summary': '__all__',
+            'time_coverage_start': '__all__',
+            'time_coverage_end': '__all__',
+            'access_constraints': '__all__'
+        }
```

Comment:
```suggestion
        fields = '__all__'
```
just like serializers,can you write it here? if yes, then it simpler.

Reply:
if yes, should be changed for all other ones

Reply:
No, it does not work like that.